src/solvers/approximate/christofides.rs

```diff
@@ -59,6 +82,41 @@ pub fn christofides<const MODE: usize>(graph: &Graph) -> Solution {
     (sum_cost, hamilton_cycle)
 }
 
+#[inline]
+fn compute_exact_matching(mst: &Graph, graph: &NAMatrix) -> Vec<[usize; 2]> {
+    // 2. compute subgraph of `graph` only with vertices that have odd degree in the MST,
+    // then compute a minimum-weight maximum matching for the subgraph
+    let subgraph: WeightedGraph = Into::<WeightedGraph>::into(graph)
+        .filter_vertices(|&vertex| mst.vertex_degree(vertex) % 2 == 1);
+
+    //note: the maximal matching is perfect
+    let matching = subgraph
+        .maximin_matching()
+        .expect("Something went wrong: could not compute the maximal minimum weight matching");
+
+    matching.edges().into_iter().map(|(a, b)| [a, b]).collect()
+}
+
+#[inline]
+fn compute_approx_matching<const MODE: usize>(mst: &Graph, graph: &NAMatrix) -> Vec<[usize; 2]> {
```

Comment:
docstring would be nice, even if you just ref to `approx_min_cost_matching`

Reply:
good suggestion